src/docfx/build/resource/BuildResource.cs

```diff
@@ -3,21 +3,42 @@
 
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.IO;
 
 namespace Microsoft.Docs.Build
 {
     internal class BuildResource
     {
-        internal static (List<Error> errors, ResourceModel model, List<string> monikers) Build(Context context, Document file)
+        internal static (List<Error> errors, PublishItem publishItem) Build(Context context, Document file)
         {
             Debug.Assert(file.ContentType == ContentType.Resource);
 
+            var outputPath = default(string);
             var (errors, monikers) = context.MonikerProvider.GetFileLevelMonikers(file, context.MetadataProvider);
-            return (errors, new ResourceModel
+
+            if (file.Docset.Config.Output.CopyResources)
+            {
+                outputPath = file.GetOutputPath(monikers);
+                context.Output.Copy(file, outputPath);
+            }
+            else
             {
+                // Output path is source file path relative to output folder when copy resource is disabled
+                outputPath = PathUtility.NormalizeFile(
```

Comment:
they can be in different disk, what about still use the absolute path?

Reply:
When it is in a different disk, `GetRelativePath` should still produce the correct path. It is defined as path relative to `.publish.json`, so the publish routine can just _combine_ output folder and `path` to locate the file.